### ISSUE TITLE
jacoco api Error while analyzing xxxx.class

### ISSUE THREAD

**Issue by bjfumeng** [jacoco/jacoco#989]
When I use java api analyzer.analyzeAll();
An error has occurred

```
java.io.IOException: Error while analyzing XXX.class.
        at org.jacoco.core.analysis.Analyzer.analyzerError(Analyzer.java:161)
        at org.jacoco.core.analysis.Analyzer.analyzeClass(Analyzer.java:133)
        at org.jacoco.core.analysis.Analyzer.analyzeClass(Analyzer.java:156)
        at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:192)
        at org.jacoco.core.analysis.Analyzer.analyzeZip(Analyzer.java:264)
        at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:195)
        at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:225)
        at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:220)
        at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:220)
        at org.jacoco.core.analysis.Analyzer.analyzeAll(Analyzer.java:220)
        at com.netease.bjxiongwenjia.webhook.user.service.login.ReportGenerator1.analyzeStructure(ReportGenerator1.java:111)
        at com.netease.bjxiongwenjia.webhook.user.service.login.ReportGenerator1.create$sentryProxy1(ReportGenerator1.java:68)
        at com.netease.bjxiongwenjia.webhook.user.service.login.ReportGenerator1.create(ReportGenerator1.java)
        at com.netease.bjxiongwenjia.webhook.user.service.login.ReportGenerator1.diff$sentryProxy2(ReportGenerator1.java:139)
        at com.netease.bjxiongwenjia.webhook.user.service.login.ReportGenerator1.diff(ReportGenerator1.java)
        at com.netease.bjxiongwenjia.webhook.user.web.api.DiffCoverageController.getUserInfo$sentryProxy2(DiffCoverageController.java:55)
        at com.netease.bjxiongwenjia.webhook.user.web.api.DiffCoverageController.getUserInfo(DiffCoverageController.java)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
        at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:133)
        at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:97)
        at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:827)
        at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:738)
        at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85)
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:967)
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:901)
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
        at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:861)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:635)
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:742)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:99)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.springframework.web.filter.HttpPutFormContentFilter.doFilterInternal(HttpPutFormContentFilter.java:109)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.springframework.web.filter.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:81)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:197)
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198)
        at org.apache.catalina.core.StandardContextValve.invoke$sentryProxy(StandardContextValve.java:96)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:496)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
        at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:803)
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:790)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1459)
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalArgumentException: null
        at org.objectweb.asm.ClassVisitor.<init>(Unknown Source)
        at org.jacoco.core.internal.flow.ClassProbesVisitor.<init>(ClassProbesVisitor.java:37)
        at org.jacoco.core.internal.flow.ClassProbesVisitor.<init>(ClassProbesVisitor.java:27)
        at org.jacoco.core.internal.analysis.ClassAnalyzer.<init>(ClassAnalyzer.java:55)
        at org.jacoco.core.analysis.Analyzer$1.<init>(Analyzer.java:96)
        at org.jacoco.core.analysis.Analyzer.createAnalyzingVisitor(Analyzer.java:93)
        at org.jacoco.core.analysis.Analyzer.analyzeClass(Analyzer.java:113)
        at org.jacoco.core.analysis.Analyzer.analyzeClass(Analyzer.java:131)
        ... 70 common frames omitted
```

**Comment by Godin** [comment]
First of all - please do not ignore our template for issues, which requires specification of used versions and presence of **complete reproducer**. In absence of these the only best guess is similar to https://stackoverflow.com/a/53917054/244993 :

```
Caused by: java.lang.IllegalArgumentException: null
        at org.objectweb.asm.ClassVisitor.<init>(Unknown Source)
```

The only known to me case of `Unknown Source` in stack traces from ASM library - is usage of its JARs that do not contain debug information and which [were removed in ASM version 6.0](https://gitlab.ow2.org/asm/asm/merge_requests/3). So you're probably using old ASM version < 6.0. However latest as of today released **JaCoCo version `0.8.5` requires ASM version `7.2`**.

**Comment by bjfumeng** [comment]
JaCoCo version：0.8.4
ASM version ：7.1
